### PR TITLE
show vacuum's map as camera

### PR DESCRIPTION
**Pull request by licheng5625**
Hello,
inspired by and **partially copied** from [project](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor) , I would like to show vacuums' map on HASS as a camera.

I have a few problems, I need support from experts. 

1. The state of the camera is always unavailbe, even though I can fetch the snapshot
image

2.  I cannot open the camera view
 image
 error is :
 `2022-01-10 05:13:24 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.17.0.1 (172.17.0.1). (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36)
`
 But I can open the link `/api/camera_proxy/camera.rockrobo_v1_7d69_vacuum_map?token=55409880 ` with browser without issue.

3. When I implement the camera entity, HA was complaining about `entity_picture` and `access_token` are missing. But I checked the [orignal project](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor) or the simplest [camera project](https://github.com/robmarkcole/HASS-rest-camera). None of them set those attributs. I set them mannuly , it could cause the issue 2.
 

3.  I hava a rockbot vacuum. I am not able to test ViomiVacuum

### CODE REVIEW

**Comment by al-one** [comment]
Thanks for your PR !
[`xiaomi_cloud_map_extractor`](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor) is an excellent custom component, and it is always updated. I also use it. Maintaining this code takes a lot of time, and I don't think we should build wheels again.

**Comment by licheng5625** [comment]
I took the parse  image part. If they could make it as a reuse library, it could be great. 

My motivation is to reduce the number of login to xiaomi server. My experience is it causes unstable and make the app on the phone very laggy. If ask me to choose one central place to integrate things about xiaomi, i choose here.

**Comment by al-one** [comment]
Maybe you can try:
```python
try:
    from custom_components.xiaomi_cloud_map_extractor.xiaomi.map_data_parser import MapDataParserXiaomi
except (ModuleNotFoundError, ImportError):
    MapDataParserXiaomi = None
```
The camera entity works only when the user installs `xiaomi_cloud_map_extractor`.

**Comment by licheng5625** [comment]
As the auther mentioned [here](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/pull/220#issuecomment-1010000898), he might create a new repo to for parsing the image. 

But if you have any time, could you help me with the issue 1 and 2 ?

thank you

File: custom_components/xiaomi_miot/camera.py [hunk 1]
```diff
@@ -596,3 +598,153 @@ async def image_source(self, **kwargs):
...
+
+class VacummMapCameraEntity(BaseSubEntity, Camera):
+    def __init__(self, parent):
+        super().__init__(parent, 'vacuum_map')
+        self._should_poll = True
+        self._supported_features |= SUPPORT_ON_OFF
+        self.parent = parent
+        self._map_raw_data = None
+    @property
+    def access_tokens(self) -> str:
```
Comment (al-one) [inline custom_components/xiaomi_miot/camera.py:611]: https://github.com/home-assistant/core/blob/2b3df46f069cb4a85635023c4327a9a2a81a09b5/homeassistant/components/camera/__init__.py#L450

`access_tokens` is recommended to be managed by HA core.

Comment (al-one) [reply custom_components/xiaomi_miot/camera.py:611]: `entity_picture` should also be managed by HA core.
And your need init the base `Camera`: `Camera.__init__(self)` before [this line](https://github.com/al-one/hass-xiaomi-miot/blob/90cb64966937f10dcd5e7f9df09e7a831aac35c5/custom_components/xiaomi_miot/camera.py#L605).

**Comment by licheng5625** [comment]
test with Rockbot v1 works 
image

**Comment by licheng5625** [comment]
intergrate with [wechat notify](https://github.com/Chiunownow/home-assistant-wechatwork-notify)
IMG_20220308_202556

**Comment by BHARATHCK** [comment]
The Map for `ijai.vacuum.v*` vacuums are not yet supported by `Xiaomi cloud map extractor`. [LINK](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/200)

Does the [map_data_parser](custom_components/xiaomi_miot/vacuum_map_extractor/xiaomi/map_data_parser.py) in this PR support those? 

Just curious. Thanks.

**Pull request closed without merging**